cmd/assetsvc/postgresql_utils.go

```diff
@@ -47,54 +57,22 @@ func newPGManager(config datastore.Config, kubeappsNamespace string) (assetManag
 	return &postgresAssetManager{m}, nil
 }
 
-func exists(current []string, str string) bool {
-	for _, s := range current {
-		if s == str {
-			return true
-		}
-	}
-	return false
-}
-
 func (m *postgresAssetManager) getAllChartCategories(namespace, repo string) ([]*models.ChartCategory, error) {
-	clauses := []string{}
-	queryParams := []interface{}{}
-	if namespace != dbutils.AllNamespaces {
-		queryParams = append(queryParams, namespace, m.GetKubeappsNamespace())
-		clauses = append(clauses, "(repo_namespace = $1 OR repo_namespace = $2)")
-	}
+	var repos []string
 	if repo != "" {
-		queryParams = append(queryParams, repo)
-		clauses = append(clauses, fmt.Sprintf("repo_name = $%d", len(queryParams)))
-	}
-	repoQuery := ""
-	if len(clauses) > 0 {
-		repoQuery = "WHERE " + strings.Join(clauses, " AND ")
+		repos = []string{repo}
 	}
-	dbQuery := fmt.Sprintf("SELECT (info ->> 'category') AS name, COUNT( (info ->> 'category')) AS count FROM %s %s GROUP BY (info ->> 'category') ORDER BY (info ->> 'category') ASC", dbutils.ChartTable, repoQuery)
+	whereQuery, whereQueryParams := m._generateWhereClause(chartQuery{namespace: namespace, repos: repos})
+	dbQuery := fmt.Sprintf("SELECT (info ->> 'category') AS name, COUNT( (info ->> 'category')) AS count FROM %s %s GROUP BY (info ->> 'category') ORDER BY (info ->> 'category') ASC", dbutils.ChartTable, whereQuery)
 
-	chartsCategories, err := m.QueryAllChartCategories(dbQuery, queryParams...)
+	chartsCategories, err := m.QueryAllChartCategories(dbQuery, whereQueryParams...)
 	if err != nil {
 		return nil, err
 	}
 	return chartsCategories, nil
 }
 
-func (m *postgresAssetManager) getPaginatedChartList(namespace, repo string, pageNumber, pageSize int) ([]*models.Chart, int, error) {
-	clauses := []string{}
-	queryParams := []interface{}{}
-	if namespace != dbutils.AllNamespaces {
-		queryParams = append(queryParams, namespace, m.GetKubeappsNamespace())
-		clauses = append(clauses, "(repo_namespace = $1 OR repo_namespace = $2)")
-	}
-	if repo != "" {
-		queryParams = append(queryParams, repo)
-		clauses = append(clauses, fmt.Sprintf("repo_name = $%d", len(queryParams)))
-	}
-	repoQuery := ""
-	if len(clauses) > 0 {
-		repoQuery = "WHERE " + strings.Join(clauses, " AND ")
-	}
+func (m *postgresAssetManager) _getPaginatedChartList(whereQuery string, whereQueryParams []interface{}, pageNumber, pageSize int) ([]*models.Chart, int, error) {
```

Comment:
I am not sure why you changed this name but please follow the current convention for naming. You don't need the underscore for telling that some method is internal since methods starting with an lowercase letter are already not exported.